File: Hash Table/2341.Maximum Number of Pairs in Array/Solution.java
class Solution {
    public int[] numberOfPairs(int[] nums) {
        int[] cnt = new int[110];
        for (int v : nums) {
            cnt[v]++;
        }
        int[] ans = new int[2];
        for (int i = 0; i < 110; i++) {
            ans[0] += cnt[i] / 2;
            ans[1] += cnt[i] % 2;
        }
        return ans;
    }
}

File: Hash Table/2341.Maximum Number of Pairs in Array/Solution.py
class Solution:
    def numberOfPairs(self, nums: List[int]) -> List[int]:
        cnt = [0] * 110
        for v in nums:
            cnt[v] += 1
        ans = [0, 0]
        for i, v in enumerate(cnt):
            ans[0] += v // 2
            ans[1] += v % 2
        return ans
